### PR TITLE
Fix typo in Controlled docstring

### PR DESCRIPTION
**Pull request by frederikwilde**
**Context:**
It seems in the past `ControlledOp` was called `ControlledOperation`. There is still a reference to this class in the docstring of the `Controlled` class.

**Description of the Change:**
Updated the docstring.

**Benefits:**
Links are not dead anymore.

**Possible Drawbacks:**
n/a

**Related GitHub Issues:**
n/a

### CODE REVIEW

**Review by albi3ro — COMMENTED** [review]
`ControlledOperation` is actually a class that is now deleted. So I think this block of code can just be removed entirely.

File: pennylane/ops/op_math/controlled.py [hunk 1]
```diff
@@ -143,15 +143,6 @@ class Controlled(SymbolicOp):
         do_queue(bool):  indicates whether the operator should be
             recorded when created in a tape context
 
-    .. note::
-        This class, ``Controlled``, denotes a controlled version of any individual operation.
-        :class:`~.ControlledOp` adds :class:`~.Operation` specific methods and properties to the
-        more general ``Controlled`` class.
```
Comment (albi3ro) [inline pennylane/ops/op_math/controlled.py:149]: Sorry about this. This section should probably stay. The stuff below paragraph about `ControlledOperation` can go.

**Comment by frederikwilde** [comment]
All good. It should be good now.

**Review by albi3ro — APPROVED** [review]
Thanks for catching this!

**Review by eddddddy — APPROVED** [review]
Thanks for this!

**Pull request merged**